### ISSUE TITLE
Add Animation Patterns system

### ISSUE THREAD

**Issue by MinusKube** [MinusKube/SmartInvs#64]
Think of and implement a pattern-like system for animations.

**Comment by Sytm** [comment]
What do you mean with pattern-like animations?
Maybe each animation frame is one pattern is what I guess. 
Or do you imagine something different?

**Comment by MinusKube** [comment]
Yes I was imagining something like that, but if you or someone have a better way to do it, that would be nice.
